packages/next/src/lib/helpers/install.ts

```diff
@@ -28,72 +27,45 @@ export function install(
   dependencies: string[] | null,
   { packageManager, isOnline, devDependencies }: InstallArgs
 ): Promise<void> {
-  /**
-   * (p)npm-specific command-line flags.
-   */
-  const npmFlags: string[] = []
-  /**
-   * Yarn-specific command-line flags.
-   */
-  const yarnFlags: string[] = []
-  /**
-   * Return a Promise that resolves once the installation is finished.
-   */
-  return new Promise((resolve, reject) => {
-    let args: string[]
-    let command = packageManager
-    const useYarn = packageManager === 'yarn'
+  let args: string[] = []
 
-    if (dependencies && dependencies.length) {
-      /**
-       * If there are dependencies, run a variation of `{packageManager} add`.
-       */
-      if (useYarn) {
-        /**
-         * Call `yarn add --exact (--offline)? (-D)? ...`.
-         */
-        args = ['add', '--exact']
-        if (!isOnline) args.push('--offline')
-        args.push('--cwd', root)
-        if (devDependencies) args.push('--dev')
-        args.push(...dependencies)
-      } else {
-        /**
-         * Call `(p)npm install [--save|--save-dev] ...`.
-         */
-        args = ['install', '--save-exact']
-        args.push(devDependencies ? '--save-dev' : '--save')
-        args.push(...dependencies)
-      }
+  dependencies ??= []
```

Comment:
Is there a reason for this change if we're still doing the `dependencies && dependencies.length > 0` check below? Maybe we can also update the `dependencies` type to strictly accept an array.

Reply:
It may be unnecessary to add the like `--save-exact` kinda args when you're just doing an install (rather than adding dependencies).

This function serves both purposes.

Reply:
My question was about the `dependencies ??= []` line

Reply:
look at line 61 - it would fail without this when `dependencies` is `null`.

Reply:
I could just move line 61 to inside the check I guess is what you're saying. yeah that's fine. changing now

Reply:
What do you think about making the `dependencies` argument require an array?

Reply:
Then you couldn't necessarily use this function to just run `npm install`

Reply:
Why not? We'd still have a `dependencies.length > 0` check right? So it'd hit the `else` block and just do `install`.

Reply:
fair enough 👍